apps/managedidentity/managedidentity_test.go

```diff
@@ -146,17 +147,26 @@ func Test_SystemAssigned_Returns_Token_Success(t *testing.T) {
 			mockClient.AppendResponse(mock.WithHTTPStatusCode(http.StatusOK), mock.WithBody(responseBody), mock.WithCallback(func(r *http.Request) {
 				localUrl = r.URL
 			}))
+			// resetting cache
+			cacheManager = storage.New(nil)
```

Comment:
This affects the whole process--all other tests--so it's best to restore the original value after this test exits

```suggestion
			before := cacheManager
			defer func() { cacheManager = before }()
			cacheManager = storage.New(nil)
```